src/c2goto/library/pthread_lib.c

```diff
@@ -483,23 +582,90 @@ __ESBMC_HIDE:;
   return 0;
 }
 
+// The pthread_key_create() function shall create a thread-specific
+// data key visible to all threads in the process.
+// source: https://linux.die.net/man/3/pthread_key_create
 int pthread_key_create(pthread_key_t *key, void (*destructor)(void *))
 {
 __ESBMC_HIDE:;
-  // TODO
-  return 0;
+  __ESBMC_atomic_begin();
+  int result = 0;
```

Comment:
Put it after the assertion. If the assertion failed, we created a variable on the stack for nothing.

Reply:
Fixed.